### PR TITLE
Unify Usage of CardPopUp

### PR DESCRIPTION
**Pull request by MattMcCherry**
Removing some duplication on how we are pushing card pop-up dispatching events. Now you should be able to use the CardPopUp component around any element that needs a popup and pass it a cardNumber and it'll handle all the popup configuration for you.

Additionally, I have removed the fanning of cards which technically fixes https://github.com/Talishar/Talishar-FE/issues/40. But I'll bring it back later :)

### CODE REVIEW

File: src/routes/game/components/elements/cardPortal/CardPortal.test.tsx [hunk 1]
```diff
@@ -0,0 +1,8 @@
+import CardPortal from './CardPortal';
```
Comment (MattMcCherry) [inline src/routes/game/components/elements/cardPortal/CardPortal.test.tsx:1]: This cardPortal component is just a renamed component as this used to be CardPopUp. But it wasn't really the one doing the pop-ing up but was just the portal to show the card(s).

File: src/routes/game/components/elements/cardPopUp/CardPopUp.tsx [hunk 2]
```diff
@@ -1,71 +1,59 @@
-import React from 'react';
```
Comment (MattMcCherry) [inline src/routes/game/components/elements/cardPopUp/CardPopUp.tsx:1]: ignore this, it's renamed to cardPortal not deleted :)

File: src/routes/game/components/elements/playerHandCard/PlayerHandCard.tsx [hunk 3]
```diff
@@ -85,80 +73,6 @@ export const PlayerHandCard = (props: HandCard) => {
...
-        yCoord: yCoord
-      })
-    );
-  };
-
-  const handleMouseLeave = () => {
-    dispatch(clearPopUp());
-  };
-
-  const degree = handSize > 1 ? lerp(-15, 15, cardIndex / (handSize - 1)) : 0;
```
Comment (MattMcCherry) [inline src/routes/game/components/elements/playerHandCard/PlayerHandCard.tsx:130]: Removed the fanning. Not sure if this fanning of cards is a great idea anyway particularly for mobile devices?

File: src/routes/game/components/elements/cardPopUp/CardPopUp.tsx [hunk 4]
```diff
@@ -1,71 +1,59 @@
...
+
+type CardPopUpProps = { 
+    children: ReactNode, 
+    cardNumber: string, 
+    containerClass?: string, 
+    onClick?: () => void, 
+    isHidden?: boolean 
+};
+
+export default function CardPopUp({ children, cardNumber, containerClass, onClick, isHidden }: CardPopUpProps) {
```
Comment (MattMcCherry) [inline src/routes/game/components/elements/cardPopUp/CardPopUp.tsx:14]: This is the component that's replaced a lot of the code. We just need a cardNumber and a child component. When wrapping the child whenever it is hovered we will see the card projected into the card portal.

e.g.
```
<CardPopUp cardNumber="UPR001">
   <span>Dromai</span>
</CardPopUp>
```

File: src/routes/game/components/elements/playerHandCard/PlayerHandCard.tsx (hunk 3 above)
Comment (LaustinSpayce) [reply src/routes/game/components/elements/playerHandCard/PlayerHandCard.tsx:130]: I liked the card fanning but I did have some people say they weren't a fan (boom boom)

I'm happy for it to be removed, we can add it back in later

**Comment by MattMcCherry** [comment]
> There is a small graphical issue where it seems the border "folds back" into itself, please see the screenshot attached for what I mean:
> 
> Screenshot 2023-01-26 at 9 15 09 PM
> 
> And DFCs are on top of each other, I tested on Firefox and Chrome.
> 
> Screenshot 2023-01-26 at 9 18 53 PM

Resolved these issues, simple css fix on the container for both cards.

File: src/routes/game/components/elements/cardPortal/CardPortal.module.css [hunk 5]
```diff
@@ -1,7 +1,10 @@
-.popUp {
+.popUpContainer {
```
Comment (MattMcCherry) [inline src/routes/game/components/elements/cardPortal/CardPortal.module.css:1]: This is the change that fixed the styling issues that you raised @LaustinSpayce. This new class is used on line 65 of the CardPortal.tsx

**Review by LaustinSpayce — APPROVED** [review]

**Pull request merged**